extensions/amp-story-interactive/1.0/utils.js

```diff
@@ -0,0 +1,25 @@
+/**
+ * Copyright 2020 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {Action} from '../../amp-story/1.0/amp-story-store-service';
+
+export const updateInteractiveStoreState = (storeService, interactive) => {
+  const update = {
+    'option': interactive.getOptionSelected(),
+    'interactiveId': this.getInteractiveId_(),
+  };
+  storeService.dispatch(Action.ADD_INTERACTIVE_REACT, update);
+};
```

Comment:
What I meant here is that we should have the `getInteractiveId_()` as a helper.

- `amp-story.js` would not need to call `getImpl()` as all you need is the URL Service and the element ID
- `amp-story.js` would just do `storeService.dispatch(Action.ADD_INTERACTIVE_REACT, {option: null, interactiveId: <what you get from the helper/util>})`
- you would not need this new `getOptionSelected()` method
- you would not need to pass references to the store and interactive instance, which keeps the code more contained and less spaghetti

Reply:
I'll move the initialization and deduplication to the interactives extension so we don't do anything from amp-story.js